### PR TITLE
ACMS-3442: Automation with modules development branch

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #[ACMS-3442](https://acquia.atlassian.net/browse/ACMS-3442)

**Proposed changes**
Execute CI job with development version of Drupal.Org modules.

**Alternatives considered**
NA
**Note:** After the review is successfully completed, the job will be removed from the CI and available only for CRON.

**Testing steps**
Follow from Ticket.

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Review by apathak18 — APPROVED** [review]

File: .github/workflows/acquia_cms_cron.yml [hunk 1]
```diff
@@ -413,3 +413,77 @@ jobs:
...
+      COMPOSER_PROCESS_TIMEOUT: 1800
+      ORCA_JOB: ${{ matrix.orca-job }}
+      CONNECTOR_ID: ${{ secrets.CONNECTOR_ID }}
+      SEARCH_UUID: ${{ secrets.SEARCH_UUID }}
+      CONNECTOR_KEY: ${{ secrets.CONNECTOR_KEY }}
+      GMAPS_KEY: ${{ secrets.GMAPS_KEY }}
+      SITESTUDIO_API_KEY: ${{ secrets.SITESTUDIO_API_KEY }}
+      SITESTUDIO_ORG_KEY: ${{ secrets.SITESTUDIO_ORG_KEY }}
+      ACMS_JOB: ${{ matrix.acms_job }}
+      JOB_TYPE: "integrated-tests"
```
Comment (deepakmishra2) [inline .github/workflows/acquia_cms_cron.yml:439]: change this name to dev-version-tests

**Comment by chandan-singh7929** [comment]
Please resolve the conflict.

File: .github/workflows/acquia_cms_cron.yml (hunk 1 above)
Comment (rajeshreeputra) [reply .github/workflows/acquia_cms_cron.yml:439]: Why?

**Review by chandan-singh7929 — APPROVED** [review]
Overall changes looks good me, just few suggestions:
- This job should run on CRON
- We should implement slack notification on failure.

File: .github/workflows/acquia_cms_cron.yml (hunk 1 above)
Comment (deepakmishra2) [reply .github/workflows/acquia_cms_cron.yml:441]: we already have multiple integrated tests running, to identify this test is running specifically for dev.
cc @vishalkhode1 @chandan-singh7929

Comment (rajeshreeputra) [reply .github/workflows/acquia_cms_cron.yml:441]: We already have dev_version_test  name for this see [here](https://github.com/acquia/acquia_cms/pull/1724/files#diff-50650aef61eb5bfa410e3ceee9a6b0ad3e6829328b3c3bfbe8377dcfc1b33dbcR448-R449).

**Review by vishalkhode1 — APPROVED** [review]

File: tests/ci/install.sh [hunk 2]
```diff
@@ -62,6 +62,15 @@ if [ -n "${ACMS_JOB}" ]; then
     composer config --unset repositories.acquia_cms_common
     composer require drupal/acquia_cms_common:2.x-dev -W
   fi
+  if [ "${ACMS_JOB}" == "dev_version_test" ]; then
+    composer config extra.composer-exit-on-patch-failure false
```
Comment (vishalkhode1) [inline tests/ci/install.sh:66]: @rajeshreeputra This adds value as string, please use this instead
```suggestion
    composer config extra.composer-exit-on-patch-failure "false" --json 
```

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**